langchain/document_loaders/parsers/schema.py

```diff
@@ -0,0 +1,48 @@
+"""Schema definition for parsers."""
+import abc
+from typing import Iterable, List
+
+from langchain.document_loaders.blob_loaders import Blob
+from langchain.schema import Document
+
+
+class BaseBlobParser(abc.ABC):
+    """Abstract interface for blob parsers.
+
+    A blob parser is provides a way to parse raw data stored in a blob into one
+    or more documents.
+
+    The parser can be composed with blob loaders, making it easy to re-use
+    a parser independent of how the blob was originally loaded.
+    """
+
+    @abc.abstractmethod
+    def lazy_parse(self, blob: Blob) -> Iterable[Document]:
```

Comment:
if we're returning generators, maybe good to use `Iterator` here? Iterator has `__next__` support.

reference: https://docs.python.org/dev/library/stdtypes.html#iterator.__next__